daemon/graphdriver/overlay/overlay.go

```diff
@@ -255,10 +255,12 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 	if err != nil {
 		return err
 	}
-	if err := idtools.MkdirAllAs(path.Dir(dir), 0700, rootUID, rootGID); err != nil {
+	root := idtools.IDPair{UID: rootUID, GID: rootGID}
+
+	if err := idtools.MkdirAllAndChown(path.Dir(dir), 0700, root); err != nil {
 		return err
 	}
-	if err := idtools.MkdirAs(dir, 0700, rootUID, rootGID); err != nil {
+	if err := idtools.MkdirAndChown(dir, 0700, root); err != nil {
```

Comment:
Looks like this line is not needed at al if the `MkdirAllAndChown()` above doesn't use `path.Dir(dir)`, but perhaps I misunderstood the logic, and it's there for a specific reason

Reply:
_(copy/paste from slack for everyone reading this)_

This seems correct to me. First call basically ensures that /var/lib/docker/overlay/ exists and is a directory (and it needs to do it recursively this is why it's MkdirAll). Second call just creates <container-id> directory under it and there's no need for recursion. Surely these two can be combined, maybe the reason was clearer error diagnostics in case something is wrong. Maybe there's something else I'm missing here, but I'd leave it as is, no point in optimization.